src/realm/array.cpp

```diff
@@ -794,16 +752,10 @@ MemRef Array::create(Type type, bool context_flag, WidthType width_type, size_t
 
 // This is the one installed into the m_vtable->finder slots.
 template <class cond>
-bool Array::find_vtable(int64_t value, size_t start, size_t end, size_t baseindex, QueryStateBase* state) const
+bool Array::find_vtable(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
```

Comment:
OK, I don't find where you are actually setting the ptr for compressed arrays find.